cli/cmd/main.go

```diff
@@ -227,6 +236,123 @@ func (c *tagsCmd) Run() error {
 	return nil
 }
 
+type simulateCmd struct {
+	Path    string   `                      help:"directory path to be iterated to search for targets within the Earthfile"               arg:"" type:"path"`
+	Targets []string `short:"t"               help:"Earthly targets pattern"                                               default:"check check-* build test test-*"`
+}
+
+func (c *simulateCmd) Run() error {
+	parser := parsers.NewEarthlyParser()
+	scanner := scanners.NewFileScanner([]string{c.Path}, parser, afero.NewOsFs())
+
+	fmt.Println("Targets: ", c.Targets)
+	// Loop through target patterns.
```

Comment:
I would rather we turn this into a separate function, which just collects a sorted array of targets.
The code here then would call that, and then iterate the returned array to run each target.